File: mefApi/Models/Membre.cs
using mefapi.Enums;
using System.ComponentModel.DataAnnotations;

namespace mefApi.Models
{
    public class Membre : BaseEntity
    {
        [Required]
        public string? Nom { get; set; }
        public bool EstActif { get; set; }
        [Required]
        public Sexe Sexe { get; set; } 
        [Required]
        public int LieuAffectationId { get; set; } 
        public Agence? LieuAffectation { get; set; } 
        public string? DateNaissance { get; set; } 
        [Required]
        public string? DateAdhesion { get; set; }
        public string? LieuNaissance { get; set; } 
        public string? Photo { get; set; } 
        public string? Telephone { get; set; }
        public string? Email { get; set; }
        //public ICollection<Mouvement>? Mouvements { get; set; }
        //public ICollection<Cotisation>? Cotisations { get; set; }
        //public ICollection<Avance>? Avances { get; set; }
        //public ICollection<Credit>? Credits { get; set; }
    }
}

File: mef/src/app/core/models/membre.ts
import { Sexe } from 'src/app/models/sexe';
import { Agence } from './agence';

export class Membre {
  id: number;
  nom: string;
  email: string;
  telephone: string;
  dateNaissance: Date;
  lieuNaissance: string;
  sexe: Sexe;
  lieuAffectation: Agence;
  dateAdhesion: Date;
  estActif: boolean;
  photo: string;

  constructor(
    id: number,
    nom: string,
    email: string,
    telephone: string,
    dateNaissance: Date,
    lieuNaissance: string,
    sexe: Sexe,
    lieuAffectation: Agence,
    dateAdhesion: Date,
    estActif: boolean,
    photo: string
  ) {
    this.id = id;
    this.nom = nom;
    this.email = email;
    this.telephone = telephone;
    this.dateNaissance = dateNaissance;
    this.lieuNaissance = lieuNaissance;
    this.sexe = sexe;
    this.lieuAffectation = lieuAffectation;
    this.dateAdhesion = dateAdhesion;
    this.estActif = estActif;
    this.photo = photo;
  }
}
